src/GafferUI/ViewportGadget.cpp

```diff
@@ -250,7 +250,11 @@ void ViewportGadget::gadgetsAt( const Imath::V2f &rasterPosition, std::vector<Ga
 		SelectionScope selectionScope( this, rasterPosition, selection, IECoreGL::Selector::IDRender );
 		const Style *s = style();
 		s->bind();
-		Gadget::doRender( s );
+
+		for( int layer = (int)Layer::Back; layer < (int)Layer::Last; ++layer )
```

Comment:
So you don't like auto here, I guess? If we want to do the loop like this, should we fix it in that other spot where it comes up too and then potentially also remove that operator override we're currently dragging along to allow `++layer` without any casts? I'll mark the other bit in the code where we do the same...

Reply:
Sorry, I thought I'd put this back to the way you had it originally. I was experimenting to see what it felt like to remove the `operator++` (no problem with auto btw). I kindof liked that removing it meant it was impossible to generate invalid values without a cast, and also that the casts were now isolated in the loop where it was easy to verify that they stayed within range. So, we should either remove `operator++` and update Gadget, or replace this with the same code as in Gadget. What do you think?